docs/design/podaffinity.md

```diff
@@ -0,0 +1,615 @@
+```
+
+The `Affinity` type is defined as follows
+
+```go
+type Affinity struct {
+	PodAffinity     *PodAffinity  `json:"podAffinity,omitempty"`
+	PodAntiAffinity *PodAntiAffinity  `json:"podAntiAffinity,omitempty"`
+}
+
+type PodAffinity struct {
+	// If the affinity requirements specified by this field are not met at
+    // scheduling time, the pod will not be scheduled onto the node.
+    // If the affinity requirements specified by this field cease to be met
+    // at some point during pod execution (e.g. due to a pod label update), the
+    // system will try to eventually evict the pod from its node.
+	// When there are multiple elements, the lists of nodes corresponding to each
+	// PodAffinityTerm are intersected, i.e. all terms must be satisfied.
+	RequiredDuringSchedulingRequiredDuringExecution []PodAffinityTerm  `json:"requiredDuringSchedulingRequiredDuringExecution,omitempty"`
+    // If the affinity requirements specified by this field are not met at
+    // scheduling time, the pod will not be scheduled onto the node.
+    // If the affinity requirements specified by this field cease to be met
+    // at some point during pod execution (e.g. due to a pod label update), the
+    // system may or may not try to eventually evict the pod from its node.
+	// When there are multiple elements, the lists of nodes corresponding to each
+	// PodAffinityTerm are intersected, i.e. all terms must be satisfied.
+	RequiredDuringSchedulingIgnoredDuringExecution  []PodAffinityTerm  `json:"requiredDuringSchedulingIgnoredDuringExecution,omitempty"`
+	// The scheduler will prefer to schedule pods to nodes that satisfy
+    // the affinity expressions specified by this field, but it may choose
+    // a node that violates one or more of the expressions. The node that is
+    // most preferred is the one with the greatest sum of weights, i.e.
+    // for each node that meets all of the scheduling requirements (resource
+    // request, RequiredDuringScheduling affinity expressions, etc.),
+    // compute a sum by iterating through the elements of this field and adding
+    // "weight" to the sum if the node matches the corresponding MatchExpressions; the
+    // node(s) with the highest sum are the most preferred.
+	PreferredDuringSchedulingIgnoredDuringExecution []WeightedPodAffinityTerm  `json:"preferredDuringSchedulingIgnoredDuringExecution,omitempty"`
+}
+
+type PodAntiAffinity struct {
+	// If the anti-affinity requirements specified by this field are not met at
+    // scheduling time, the pod will not be scheduled onto the node.
+    // If the anti-affinity requirements specified by this field cease to be met
+    // at some point during pod execution (e.g. due to a pod label update), the
+    // system will try to eventually evict the pod from its node.
+	// When there are multiple elements, the lists of nodes corresponding to each
+	// PodAffinityTerm are intersected, i.e. all terms must be satisfied.
```

Comment:
These type names aren't part of the API, so we'll need to rework these comments in the real implementation.